router/internal/expr/expr.go

```diff
@@ -62,10 +55,22 @@ type RequestURL struct {
 	Query map[string]string `expr:"query"`
 }
 
+type RequestHeaders struct {
+	Header http.Header `expr:"-"` // Do not expose the full header
+}
+
+// Get returns the value of the header with the given key. If the header is not present, an empty string is returned.
+// The key is case-insensitive and transformed to the canonical format.
+func (r RequestHeaders) Get(key string) string {
+	return r.Header.Get(key)
+}
```

Comment:
I genuinely think that having an interface here would be the cleanest and most Go style solution. I understand that we have to wait for the expr team to implement the fix but let's flag this with a TODO and the link to the Github issue so that we remember to update the code in the future.

What do you think?

Reply:
Sounds good to me.